### PR TITLE
Add cors origin filter.

### PR DESCRIPTION
**Pull request by ruiaraujo**
Related to #559.

The rest of the headers can be set using setResponseHeader.

### CODE REVIEW

**Comment by szuecs** [comment]
Can you register the filter at https://github.com/zalando/skipper/blob/master/filters/builtin/builtin.go#L54 and add valid godoc documentation?
Valid godoc starts for all function docs with the function name.

    // Foo makes it easy to do ...
    func (f *F) Foo() {}

File: filters/cors/cors.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+type filter struct {
+	allowedOrigins []string
+}
+
+func NewOrigin() filters.Spec {
+	return &basicSpec{}
+}
+
+// We check for the origin header if there are allowed origins
```
Comment (szuecs) [inline filters/cors/cors.go]: We check -> Response checks...

File: filters/cors/cors.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	}
+	for _, o := range a.allowedOrigins {
+		if o == origin {
+			ctx.Response().Header.Add(allowOriginHeader, o)
+			return
+		}
+	}
+}
+
+// We do not touch request at all
```
Comment (szuecs) [inline filters/cors/cors.go]: We do -> Request does ..

File: filters/cors/cors.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+func (a filter) Request(filters.FilterContext) {}
+
+// Creates out the cors filter
+func (spec basicSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
```
Comment (szuecs) [inline filters/cors/cors.go]: Creates -> CreateFilter ..

**Comment by ruiaraujo** [comment]
👍

**Comment by aryszka** [comment]
:+1:

**Pull request merged**